slang-gfx.h

```diff
@@ -435,11 +435,13 @@ enum class MemoryType
 enum class InteropHandleAPI
 {
     Unknown,
-    D3D12,
-    Vulkan,
-    CUDA,
-    Win32,
-    FileDescriptor,
+    D3D12, // A D3D12 object pointer.
+    Vulkan, // A general Vulkan object handle.
+    CUDA, // A general CUDA object handle.
+    Win32, // A general Win32 HANDLE.
+    FileDescriptor, // A file descriptor.
+    DeviceAddress, // A device address.
+    D3D12CpuDescriptorHandle, // A D3D12_CPU_DESCRIPTOR_HANDLE value.
```

Comment:
Unrelated to the meat of the PR:

Okay.... I've said it before and I'm going to say it again: the `InteropHandleAPI` should never have been named/described as an enumeration of different APIs. It is an enumeration of different *types* or *formats* for handles.

The cases in `InteropHandleType` (or `InteropHandleFormat`) could, if desired, be stored in a bit representation where the actual API is the high-order bits and then the low-order bits hold the API-specific type/format:

```
enum class InteropPlatform
{
    D3D12,
    Vulkan,
    Win32,
    POSIX,
    ...
};
enum class InteropHandleType
{
    D3D12_IResource = (uint(InteropPlatform::D3D12) << 16),
    D3D12_IShaderResourceView,
    D3D12_ISwapChain,
    D3D12_CpuDescriptorHandle,
    // ...

    Win32_HANDLE = (uint(InteropPlatform::Windows) << 16),
};
```

Obviously we could decide not to bother with that kind of scoping, and just have a flat enum that lists all the `<API>_<HandleType>` pairs that matter in arbitrary order. The key point is that the entries here should really all be in the form `<API>_<HandleType>`.

Reply:
Yes, we should be implementing this refactor sooner than later.